kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/AllowLaunchSuspendFunctionFromThread.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2010-2022 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.konan
+
+enum class AllowLaunchSuspendFunctionFromThread {
```

Comment:
The name and the descriptions below are a bit misleading: this flag covers only the case when calling a Kotlin `suspend` function from Swift or Objective-C.
When calling from Kotlin, there are no such limitations, and the flag does not affect this.

Same for other similar names. Same for the commit message.

Reply:
changed - happy to change this if you may recommend a better suggestion!

Reply:
Thanks!

The new version is still not entirely correct -- even on Apple targets, there were no thread restrictions when calling `suspend` function from Kotlin.
It is not the target that matters with the existing restriction, but the fact that `suspend` function is called from Objective-C (or Swift).

What do you think about something like `ObjCExportSuspendFunctionLaunchThreadRestriction`?

Reply:
done.

Reply:
sounds good, will do!

Reply:
done